### PR TITLE
Tanja and Shaunna -- Scrabble -- Carets 

### PR DESCRIPTION
**Pull request by misstonbon**
# Scrabble
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Feature 	|   Feedback	|
|---	|---	|
|   Why do you think we chose to make the `score` method in the `Scoring` class a **class method** instead of an **instance method**? | We did not need an instance variable for Scoring because the Scoring class' only job is to calculate the word score.    |
|   Describe an example of a test you wrote for a _nominal case_. |   We tested that the words are scored properly. For example, for the word "seven, the score is 8.	|
|   Describe an example of a test you wrote for an _edge case_. | Score must be 0 for empty array, instead of raising an error. Also, when no more tiles are left, and no one player scored 100, instead of raising an error we have the program return an empty array.  	|
|   What was your final test coverage percentage?	|   98 %	|
|   Is there a specific piece of code you'd like feedback on?	|   Error handling .	|

### CODE REVIEW

File: lib/wave-1-scoring.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      letters = word.upcase.split("")
+      word_score = 0
+
+      letters.each do |letter|
+        SCORE_HASH.each do |letter_arr, letter_score|
+          if letter_arr.include? letter
+            word_score += letter_score
+          end
+        end
+      end
```
Comment (CheezItMan) [inline lib/wave-1-scoring.rb:28]: this could probably be:

```
letters.each do |letter|
  SCORE_HASH.keys.each do |key|
    if key.include? letter
      word_score += SCORE_HASH[key]
    end
  end
end
```

**Comment by CheezItMan** [comment]
# Scrabble
## What We're Looking For

|  Feature  |   Feedback |
|---	|---	|
|  **General**	|   	|
|   Answered comprehension questions  | Check, for error handling we'll have something on Exceptions on Wednesday.   |
|   Both Teammates contributed to the codebase	|  Check, well done  |
|   Regular Commits with meaningful commit messages  |  Good commit messages and a good number of commits.    |
|  **Scoring class**	|   	|
|  `score` method	|  Check  |
|   Uses appropriate data structure to store the letter score	|  Check, good use of the hash.  |
|   Appropriately handles edge cases	|  Well done!  |
|   Tests `score` edge cases (one-letter word, 7 letter word)	|  Check  |
|  `highest_scoring_word` method	|  Lots of enumerable use here, it is a bit long however.  You could either break this up a bit into another helper method or simplify it further with enumerables.    |
|   Appropriately handles edge cases	|  Check for 7-letter words, and handles 1 word arrays and checks for empty arrays.  |
|   Tests `highest_scoring_word` edge cases (ties, no words, no array)	|  I would check that if the order of the words vary the answer is still proper.  |
|  **Player class**	|   	|
|  Uses an array to keep track of words played	|  Check  |
|  Uses existing code to create `highest_scoring_word` and `highest_word_score`	|  Check   |
|  Returns `true` or `false` from `won?` method | Check, this could simply have been `return total_core > 100`  |
|  Tests edge cases on the `play(word)` method for words that cannot be scored |  Check   |
|  **TileBag class**	|   	|
|  Uses a data structure to keep the set of default tiles in the bag	|  Check, well done  |
|  `draw_tiles` method uses the parameter to draw a set of random tiles |  Check  |
| `tiles_remaining` method returns a count	|  Check  |
|  Tests `draw_tiles` edge cases for different parameter inputs	|  Interesting choice when they try to draw more tiles than remain, it returns `[]`.  Why not return the few remaining tiles?    |
|  Summary  | Well done, you hit all the requirement.  Awesome work!  Impressive work on the board!   |

**Review by CheezItMan — COMMENTED** [review]
Well done!
